contrib/init/systemd/docker.service

```diff
@@ -1,34 +1,47 @@
 [Unit]
 Description=Docker Application Container Engine
 Documentation=https://docs.docker.com
-After=network-online.target docker.socket firewalld.service
+After=network-online.target docker.socket firewalld.service containerd.service
 Wants=network-online.target
-Requires=docker.socket
+Requires=docker.socket containerd.service
 
 [Service]
 Type=notify
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
-ExecStart=/usr/bin/dockerd -H fd://
+ExecStart=/usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
 ExecReload=/bin/kill -s HUP $MAINPID
-LimitNOFILE=1048576
+TimeoutSec=0
```

Comment:
Actually; I should probably pick the one from this repository; looks like `TimeoutSec` is deprecated in favour of `TimeoutStartSec`; https://github.com/systemd/systemd/issues/11155

Reply:
Changed this one to `TimeoutStartSec`